components/org.wso2.identity.webhook.wso2.event.handler/src/main/java/org/wso2/identity/webhook/wso2/event/handler/builder/WSO2LoginEventPayloadBuilder.java

```diff
@@ -17,53 +17,207 @@
+        String errorCode = dataMap.get(Constants.CURRENT_AUTHENTICATOR_ERROR_CODE);
+        failedReason.setId(errorCode);
+
+        AuthenticationFailedReason.FailedStep failedStep = new AuthenticationFailedReason.FailedStep();
+        failedStep.setStep(authContext.getCurrentStep());
+        failedStep.setAuthenticator(authContext.getCurrentAuthenticator());
+        failedStep.setIdp(authContext.getExternalIdP() != null ?
+                authContext.getExternalIdP().getIdentityProvider().getIdentityProviderName() : null);
+        failedReason.setFailedStep(failedStep);
+
+        return failedReason;
+    }
+
+    private void populateUserAttributes(AuthenticatedUser authenticatedUser, User user) {
+
+        if (authenticatedUser == null) {
+            return;
+        }
+
+        List<UserClaim> userClaims = new ArrayList<>();
+        for (Map.Entry<ClaimMapping, String> entry : authenticatedUser.getUserAttributes().entrySet()) {
+            ClaimMapping claimMapping = entry.getKey();
+            String claimUri = claimMapping.getLocalClaim().getClaimUri();
+            String claimValue = entry.getValue();
+
+            if (claimUri != null && claimValue != null) {
+                switch (claimUri) {
+                    case Constants.WSO2_CLAIM_GROUPS:
+                        user.addGroup(claimValue);
+                        break;
+                    case Constants.WSO2_CLAIM_ROLES:
+                        user.addRole(claimValue);
+                        break;
+                    case Constants.MULTI_ATTRIBUTE_SEPARATOR:
+                        // Not adding the multi attribute separator to the user claims
+                        break;
+                    case Constants.USER_ORGANIZATION:
+                        // Not adding the user resident organization to the user claims for b2b users
+                        break;
+                    default:
+                        userClaims.add(new UserClaim(claimUri, claimValue));
+                        break;
+                }
+            }
+        }
+        user.setClaims(userClaims);
+    }
+
+    private Organization getUserResidentOrganization(String organizationId) throws IdentityEventException {
+
+        try {
+            String organizationName = WSO2EventHookHandlerDataHolder.getInstance()
+                    .getOrganizationManager().getOrganizationNameById(organizationId);
+            return new Organization(organizationId, organizationName);
+        } catch (OrganizationManagementException e) {
+            if (ERROR_CODE_INVALID_ORGANIZATION_ID.getCode().equals(e.getErrorCode())) {
+                log.debug("Returning an empty string as the organization name as the name is not returned for the given id.");
+            }
+            throw new IdentityEventException(UNEXPECTED_SERVER_ERROR.getCode(),
+                    "Error while retrieving the organization name for the given id.", e);
+        }
```

Comment:
Better not throw and break the flow.
Given these are events, it's not significant that the organization name is not present

Reply:
Addressed with https://github.com/Sachin-Mamoru/identity-webhook-event-handlers/pull/1/commits/8750eff9ee3a84ff76fd11671bbbc3451fbb0fc8